### PR TITLE
test(stackdriver): add tracing integration test for VMs

### PR DESCRIPTION
**Pull request by douglas-reid**
PR to add coverage for tracing to Stackdriver integration tests.  Refactors the test structure a bit in an attempt to improve readability by separating out the tests parts and the setup parts (as well as to fail faster when there are setup issues).

Notes:
- a custom bootstrap override is used for the VM to enable population of a testing endpoint for Cloud Trace (it is not read from env vars, as elsewhere)
- a custom proxy config is used to simulate real-world VM configuration reqs for Cloud Trace.
- `custom_tags` are added to each span to simulate annotation of canonical service information with each span.

[ X ] Policies and Telemetry
[ X ] Test and Release
[ X ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/telemetry/stackdriver/vm/main_test.go [hunk 1]
```diff
@@ -59,7 +127,14 @@ func TestMain(m *testing.M) {
...
 		NewSuite(m).
 		RequireSingleCluster().
-		Setup(istio.Setup(&i, func(_ resource.Context, cfg *istio.Config) {
+		Setup(istio.Setup(&istioInst, func(_ resource.Context, cfg *istio.Config) {
+			cfg.Values["meshConfig.enableTracing"] = "true"
+			cfg.Values["meshConfig.defaultConfig.tracing.sampling"] = "100.0"
+			cfg.Values["meshConfig.defaultConfig.tracing.custom_tags.canonical_service_name.environment.name"] = "CANONICAL_SERVICE"
+			cfg.Values["meshConfig.defaultConfig.tracing.custom_tags.canonical_service_name.environment.defaultValue"] = "unknown"
+			cfg.Values["meshConfig.defaultConfig.tracing.custom_tags.canonical_service_revision.environment.name"] = "CANONICAL_REVISION"
+			cfg.Values["meshConfig.defaultConfig.tracing.custom_tags.canonical_service_revision.environment.defaultValue"] = "latest"
```
Comment (gargnupur) [inline tests/integration/telemetry/stackdriver/vm/main_test.go:136]: should this be set to `earliest` too as in envTagsProxyConfig?

Comment (douglas-reid) [reply tests/integration/telemetry/stackdriver/vm/main_test.go]: great catch. updated.

File: tests/integration/telemetry/stackdriver/vm/main_test.go [hunk 2]
```diff
@@ -113,7 +189,160 @@ func testSetup(ctx resource.Context) (err error) {
...
+	clientBuilder = echoboot.NewBuilder(ctx).
+		With(&client, echo.Config{
+			Service:   "client",
+			Namespace: ns,
+			Ports:     ports,
+			Subsets: []echo.SubsetConfig{
+				{
+					Annotations: map[echo.Annotation]*echo.AnnotationValue{
+						echo.SidecarBootstrapOverride: {
+							Value: sdBootstrapConfigMap,
```
[the remark this answers is not in the document]
Comment (douglas-reid) [reply tests/integration/telemetry/stackdriver/vm/main_test.go]: good idea. done.

**Comment by douglas-reid** [comment]
@gargnupur thanks for the review! updated to address your comments. ptal.

File: tests/integration/telemetry/stackdriver/vm/main_test.go [hunk 3]
```diff
@@ -113,7 +189,156 @@ func testSetup(ctx resource.Context) (err error) {
...
+	if err != nil {
+		return
+	}
+	cr, err := tmpl.Evaluate(string(cltRequestCountTmpl), map[string]interface{}{
+		"EchoNamespace": ns.Name(),
+	})
+	if err != nil {
+		return
+	}
+	err = jsonpb.UnmarshalString(cr, cltRequestCount)
```
Comment (gargnupur) [inline tests/integration/telemetry/stackdriver/vm/main_test.go:285]: nit: move declaration of cltRequestCount here.. so it's closer to where it's getting used?

File: tests/integration/telemetry/stackdriver/vm/vm_test.go [hunk 4]
```diff
@@ -45,222 +42,150 @@ func TestVMTelemetry(t *testing.T) {
...
 	}
-	err = jsonpb.UnmarshalString(cr, &cltRequestCount)
-	return
+
+	return gotServer && gotClient
 }
 
-func getWantServerLogEntry() (srvLogEntry loggingpb.LogEntry, err error) {
-	srvlogEntryTmpl, err := ioutil.ReadFile(serverLogEntry)
+func gotLogEntry(want *loggingpb.LogEntry) bool {
```
Comment (gargnupur) [inline tests/integration/telemetry/stackdriver/vm/vm_test.go:131]: for later.. we should combine these with non vm tests too?

**Review by gargnupur — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: tests/integration/telemetry/stackdriver/vm/vm_test.go (hunk 4 above)
Comment (douglas-reid) [reply tests/integration/telemetry/stackdriver/vm/vm_test.go:131]: Yes. probably should try to harmonize both sets of tests.

**Pull request merged**